src/clr-angular/data/datagrid/render/header-renderer.spec.ts

```diff
@@ -93,7 +99,190 @@ export default function(): void {
       expect(context.clarityElement.classList).not.toContain('datagrid-fixed-width');
     });
   });
+
+  describe('Datagrid Header Resize Rendering', function() {
+    let context: TestContext<ClrDatagrid, HeaderResizeTestComponent>;
+
+    let columnHeader1DebugElement: DebugElement;
+    let columnHeader2DebugElement: DebugElement;
+    let columnHeader3DebugElement: DebugElement;
+    let columnHeader4DebugElement: DebugElement;
+
+    let columnHeader1Element: HTMLElement;
+    let columnHeader2Element: HTMLElement;
+    let columnHeader3Element: HTMLElement;
+    let columnHeader4Element: HTMLElement;
+
+    let column1InitialWidth: number;
+    let column2InitialWidth: number;
+    let column3InitialWidth: number;
+    let column4InitialWidth: number;
+
+    let columnHeader1ResizerService: ColumnResizerService;
+    let columnHeader2ResizerService: ColumnResizerService;
+    let columnHeader3ResizerService: ColumnResizerService;
+    let columnHeader4ResizerService: ColumnResizerService;
+
+    let columnHeader1DraggableDebugElement: DebugElement;
+    let columnHeader3DraggableDebugElement: DebugElement;
+
+    let columnHeader1DraggableDirective: ClrDraggable<any>;
+    let columnHeader3DraggableDirective: ClrDraggable<any>;
+
+    const widthOf = (el: HTMLElement): number => {
+      return Math.round(el.getBoundingClientRect().width);
+    };
+
+    const emulateResizeOnColumn = (moveX: number, columnDraggable: ClrDraggable<any>): void => {
+      const moveEvent = new ClrDragEvent({
+        type: DragEventType.DRAG_MOVE,
+        dragPosition: { pageX: 0, pageY: 0, moveX: moveX, moveY: 0 },
+      });
+      columnDraggable.dragStartEmitter.emit();
+      columnDraggable.dragMoveEmitter.emit(moveEvent);
+      columnDraggable.dragEndEmitter.emit();
+    };
+
+    beforeEach(function() {
+      context = this.create(ClrDatagrid, HeaderResizeTestComponent);
+
+      columnHeader1DebugElement = context.fixture.debugElement.queryAll(By.directive(DatagridHeaderRenderer))[0];
+      columnHeader2DebugElement = context.fixture.debugElement.queryAll(By.directive(DatagridHeaderRenderer))[1];
+      columnHeader3DebugElement = context.fixture.debugElement.queryAll(By.directive(DatagridHeaderRenderer))[2];
+      columnHeader4DebugElement = context.fixture.debugElement.queryAll(By.directive(DatagridHeaderRenderer))[3];
+
+      columnHeader1ResizerService = columnHeader1DebugElement.injector.get(ColumnResizerService);
```

Comment:
Does there need to be one of these for each column? What happens if there is just one columnHeaderResizerService and all of the columns use it?